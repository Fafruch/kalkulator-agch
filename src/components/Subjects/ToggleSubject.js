import React from 'react'
import PropTypes from 'prop-types'

const ToggleSubject = ({ id, active, subjectsType, onToggleClick }) => (
  <img
    src={active ? 'icons/toggle-on-96.png' : 'icons/toggle-off-96.png'}
    alt={active ? 'Nie bierz pod uwagę przedmiotu' : 'Bierz przedmiot pod uwagę'}
    onClick={() => onToggleClick(id, subjectsType)}
    className='subject-icon'
  />
)
ToggleSubject.propTypes = {
  id: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired,
  subjectsType: PropTypes.string.isRequired,
  onToggleClick: PropTypes.func.isRequired
}

export default ToggleSubject
