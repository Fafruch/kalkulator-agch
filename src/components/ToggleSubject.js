/**
 * Created by Fafruch on 17/07/2017.
 */

import React from 'react'
import PropTypes from 'prop-types'

const ToggleSubject = ({ id, active, onToggleClick }) => (
  <img
    src={active ? 'https://png.icons8.com/toggle-on/win10/96' : 'https://png.icons8.com/toggle-off/win10/96'}
    alt={active ? 'Nie bierz pod uwagę przedmiotu' : 'Bierz przedmiot pod uwagę'}
    style={{ width: 32, height: 32 }}
    onClick={() => onToggleClick(id)}
  />
)
ToggleSubject.propTypes = {
  id: PropTypes.number.isRequired,
  active: PropTypes.bool.isRequired,
  onToggleClick: PropTypes.func.isRequired
}

export default ToggleSubject
