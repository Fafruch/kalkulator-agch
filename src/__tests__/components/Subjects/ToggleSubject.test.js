import React from 'react'
import { shallow } from 'enzyme'

import ToggleSubject from '../../../components/Subjects/ToggleSubject'

const setup = (setupProps = {}) => {
  const defaultProps = {
    id: '0',
    active: true,
    subjectsType: 'primary',
    onToggleClick: jest.fn()
  }
  const props = { ...defaultProps, ...setupProps }
  const wrapper = shallow(<ToggleSubject {...props} />)

  return {
    props,
    wrapper
  }
}

describe('ToggleSubject component', () => {
  test('should render', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })

  test('should render toggled-off icon when subject is not active', () => {
    const { wrapper } = setup({ active: false })

    expect(wrapper.prop('src')).toEqual('/icons/toggle-off-96.png')
  })

  test('should have proper alt attribute based on activeness', () => {
    const { wrapper } = setup({ active: false })

    expect(wrapper.prop('alt')).toEqual('Bierz przedmiot pod uwagÄ™')
  })

  test('should call onToggleClick when clicked on a button', () => {
    const { props, wrapper } = setup()

    wrapper.simulate('click')
    expect(props.onToggleClick).toBeCalled()
  })
})
