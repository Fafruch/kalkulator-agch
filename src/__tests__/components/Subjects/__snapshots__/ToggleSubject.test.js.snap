// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSubject component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToggleSubject
    active={true}
    id="0"
    onToggleClick={[Function]}
    subjectsType="primary"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "alt": "Nie bierz pod uwagę przedmiotu",
      "className": "subject-icon",
      "onClick": [Function],
      "src": "icons/toggle-on-96.png",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "alt": "Nie bierz pod uwagę przedmiotu",
        "className": "subject-icon",
        "onClick": [Function],
        "src": "icons/toggle-on-96.png",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
